### ISSUE TITLE
Use controlled components in tutorial

### ISSUE THREAD

**Issue by zpao** [facebook/react#5443]
Based on some discussions had outside of github, we're going to move away from the use of refs in the tutorial and use controlled components instead. This is the most "reacty" way to do what is done in the tutorial and avoids the escape hatch of refs.

**Comment by yangshun** [comment]
My attempt at rewriting the `CommentForm` using controlled components:

```
var CommentForm = React.createClass({
  getInitialState: function() {
    return {author: '', text: ''};
  },
  handleChange: function(field, e) {
    var newState = {};
    newState[field] = e.target.value;
    this.setState(newState);
  },
  handleSubmit: function(e) {
    e.preventDefault();
    var author = this.state.author.trim();
    var text = this.state.text.trim();
    if (!text || !author) {
      return;
    }
    this.props.onCommentSubmit({author: author, text: text});
    this.setState({author: '', text: ''});
  },
  render: function() {
    return (
      <form className="commentForm" onSubmit={this.handleSubmit}>
        <input type="text" placeholder="Your name" value={this.state.author} onChange={this.handleChange.bind(this, 'author')}/>
        <input type="text" placeholder="Say something..." value={this.state.text} onChange={this.handleChange.bind(this, 'text')}/>
        <input type="submit" value="Post" />
      </form>
    );
  }
});
```

Not too sure doing `this.handleChange.bind` is a good idea. Any suggestions? @jimfb @zpao

**Comment by zpao** [comment]
I agree. I think I might try to avoid bind for the purposes of the tutorial and just have 2 explicit handlers.

**Comment by yangshun** [comment]
ok I'll make a PR :smile:

**Comment by zpao** [comment]
Awesome. It's going to require more than just the code change - we want to make sure the concept is adequately explained as well. Looking forward to it!

**Comment by yangshun** [comment]
Fixed by #5445

**Comment by jimfb** [comment]
Great, thanks!
